.github/workflows/ci.yml

```diff
@@ -22,13 +22,20 @@ jobs:
       # and hyphens.
       run: echo "TAG=branch-${GITHUB_REF_NAME//[^A-Za-z0-9._-]/-}" | tee -a $GITHUB_ENV
 
+    - if: ${{ github.event_name != 'pull_request' }}
+      name: Set $DIRECT_PUSH=true
+      run: echo "DIRECT_PUSH=true" | tee -a $GITHUB_ENV
+    - if: ${{ github.event_name == 'pull_request' }}
+      name: Set $DIRECT_PUSH=false
+      run: echo "DIRECT_PUSH=false" | tee -a $GITHUB_ENV
+
     - if: ${{ github.event_name != 'pull_request' }}
       uses: docker/login-action@v1
       with:
         username: ${{ secrets.DOCKER_LOGIN }}
         password: ${{ secrets.DOCKER_PASSWORD }}
 
-    - run: ./devel/build $TAG
+    - run: ./devel/build $TAG $DIRECT_PUSH
```

Comment:
This logic in the workflow and the additional logic required in `./devel/build` seems to me to be making things more complex than they need to be. The intent of the programs in `./devel/…` is to be composable and easily usable both by CI and local dev. Making them do multiple things is at cross-purposes with that.

Instead of making `./devel/build` optionally push, what about using the existing `./devel/push` instead? e.g. get rid of the `DIRECT_PUSH` var and the changes to `./devel/build` and instead do something like:

```yaml
- if: github.event_name != 'pull_request'
  run: ./devel/push $TAG
```

after the `./devel/build` step here.

Reply:
Thanks for catching me here, I just reworked it and am happier with the new approach.

Still keeping the option to provide a tag to `./devel/build` since I find the current usage of `latest` to be a bit complex.